Source/Scene/TileBoundingS2Cell.js

```diff
@@ -0,0 +1,695 @@
+   */
+  boundingVolume: {
+    get: function () {
+      return this;
+    },
+  },
+  /**
+   * The underlying bounding sphere.
+   *
+   * @memberof TileOrientedBoundingBox.prototype
+   *
+   * @type {BoundingSphere}
+   * @readonly
+   */
+  boundingSphere: {
+    get: function () {
+      return this._boundingSphere;
+    },
+  },
+});
+
+var facePointScratch = new Cartesian3();
+/**
+ * The distance to point check for this kDOP involves checking the signed distance of the point to each bounding
+ * plane. A plane qualifies for a distance check if the point being tested against is in the half-space in the direction
+ * of the normal i.e. if the signed distance of the point from the plane is greater than 0.
+ *
+ * There are 4 possible cases for a point:
+ *
+ *   \     X     /     X \           /       \           /       \           /
+ * ---\---------/---   ---\---------/---   ---X---------/---   ---\---------/---
+ *     \       /           \       /           \       /           \       /
+ *   ---\-----/---       ---\-----/---       ---\-----/---       ---\-----/---
+ *       \   /               \   /               \   /               \   /
+ *                                                                    \ /
+ *                                                                     \
+ *                                                                    / \
+ *                                                                   / X \
+ *
+ *         I                  II                  III                 IV
+ *
+ * Case I: There is only one plane selected.
+ * In this case, we project the point onto the plane and do a point polygon distance check to find the closest point on the polygon.
+ * The point may lie inside the "face" of the polygon or outside. If it is outside, we need to determine which edges to test against.
+ *
+ * Case II: There are two planes selected.
+ * In this case, the point will lie somewhere on the line created at the intersection of the selected planes.
+ *
+ * Case III: There are three planes selected.
+ * In this case, the point will lie on the vertex, at the intersection of the selected planes.
+ *
+ * Case IV: There are more than three planes selected.
+ * Since we are on an ellipsoid, this will only happen in the bottom plane, which is what we will use for the distance test.
+ */
+TileBoundingS2Cell.prototype.distanceToCamera = function (frameState) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.defined("frameState", frameState);
+  //>>includeEnd('debug');
+
+  var point = frameState.camera.positionWC;
+
+  var selectedPlaneIndices = [];
+  var vertices;
+  var edgeNormals;
+  var rotation;
+
+  if (
+    CesiumMath.greaterThan(
+      Plane.getPointDistance(this._boundingPlanes[0], point),
+      0,
+      CesiumMath.EPSILON14
+    )
+  ) {
+    selectedPlaneIndices.push(0);
+    vertices = this._vertices.slice(0, 4);
+    edgeNormals = this._edgeNormals[0];
+    rotation = -1;
```

Comment:
Took me a while to realize that `rotation` is for flipping the edge normals when testing if the point is in the polygon faces. Though which direction should the normals point for that test? inwards or outwards? I'm a little confused.

Reply:
Rotations make the code confusing, so I refactored the code to not need this.